### PR TITLE
Support multiple languages

### PR DESCRIPTION
**Pull request by doomspork**
Here it is @erickgnavar, @ZuraGuerra, and @3100.  This pull request is first draft at implementing multiple language support on elixirschool.com.

To create a new language:

1. Create a folder for the 2 character language code (e.g. jp, en, es, etc)
1. Add `lessons/{basics,advanced,specifics}` folders inside it.
1. Move translated lessons over.
1. Add `lang: XX` to the lesson page meta data.
1. Add the language to _config.yml as part of `languages` and the `sections` translations.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -9,9 +9,22 @@ author:
...
-url: 'http://elixirschool.com' # the base hostname & protocol for your site
+baseurl: /
+url: https://elixirschool.com
 permalink: pretty
+languages: ['en', 'jp']
+default_lang: en
+sections:
+  - tag: basics
+    label:
+      en: Basics
```
Comment (doomspork) [inline _config.yml]: I thought it might be useful to translate the section headers.

File: index.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 layout: page
 title: Elixir School
+lang: en
```
Comment (doomspork) [inline index.md]: Every lesson will need a `lang` tag.  This is used generating the sidebar.

**Comment by 3100** [comment]
This is awesome! Great work :+1: 
I [tried](https://github.com/3100/elixir-school/commit/343030d900b521e155610d93d3ce06890cbcf478) adding some Japanese lessons and it worked.  There seem to be some improvements:

- Langages' links (e.g. `jp`) below the sidebar redirect us to https://elixirschool.com and it's a bit inconveninent when debugging. (Other links such as lessons are relative.)
- The sections are not translated in my environment.

**Comment by doomspork** [comment]
Sorry @3100 I should have addressed your first point in my original PR comments:

- As a work around to how Jekyll handles paths I'm relying on the `url` value in `_config.yml` to provide the base for assembling the language links.  For me locally, the value `http://localhost:4000` does the trick.  I agree there could be a better solution than repeatedly changing that value.  I'll keep at it :grinning: 

- Could you be more specific when you say the sections are not translated?  Do you mean the section headers (Basics, Advanced, Specifics) or the actual lessons themselves?

Thanks for your help!

File: _config.yml [hunk 3]
```diff
@@ -9,9 +9,22 @@ author:
 # Setup
 title: Elixir School
 description: Lessons in the Fundamentals of Elixir
-baseurl: '/'
-url: 'http://elixirschool.com' # the base hostname & protocol for your site
+baseurl: /
+url: https://elixirschool.com
```
Comment (doomspork) [inline _config.yml]: @3100 this configuration value is used in the generation of the language links.  I'll see if there's a way to assign values for development vs production in Jekyll.

File: _includes/sidebar.html [hunk 4]
```diff
@@ -2,34 +2,47 @@
...
               </div>
             {% endfor %}
           </div>
         </div>
       {% endfor %}
-
     </nav>
 
+    <p>
+      {% if page.lang == site.default_lang %}
```
Comment (doomspork) [inline _includes/sidebar.html]: @3100 this is the messy part with the language URLs and why I ultimately went with `site.url` to solve it.

When viewing a translated lesson the `page.url` would be something akin to `/jp/lessons/basics/mix/` and relative links weren't doing the trick.  I also wanted to ensure the language links would go to the current lesson albeit translated:

1. If `page.lang` is set to something other than our default, remove the language code from the beginning of the `page.url`.  This leaves us with the section and lesson we're viewing. 

      > /jp/lessons/basics/mix -> /lessons/basics/mix

1. To create the links I concat `site.url` (our config value), `lang`, and the `url` from Step 1.

**Comment by 3100** [comment]
@doomspork Thanks for your information and now I see how to config languages' link paths on the sidebar for development. 

> - Could you be more specific when you say the sections are not translated? Do you mean the section headers (Basics, Advanced, Specifics) or the actual lessons themselves?

Sorry, I mean the section headers.

**Comment by doomspork** [comment]
I'll look at the section headers this morning for you @3100 :grinning:

**Comment by doomspork** [comment]
@3100 I pushed a change to the PR, could you take a look?  You'll need to update the `_config.yml`:

```yml
sections:
  - tag: basics
    label:
      en: Basics
      jp: 目次
  - tag: advanced
    label:
      en: Advanced
      jp: 目次
  - tag: specifics
    label:
      en: Specifics
      jp: 目次
```

**Comment by 3100** [comment]
@doomspork Thanks, now it translates perfectly :+1:

**Comment by doomspork** [comment]
That's great news @3100!  

I'm going to poke at this a bit more over the weekend and then deploy these changes.  After that we can start getting translations online.

**Comment by doomspork** [comment]
Thanks again for your help @3100! :+1:

**Pull request merged**